sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.institutions.controller.getRequestedAssessmentSummary.e2e-spec.ts

```diff
@@ -106,6 +107,53 @@ describe("AssessmentInstitutionsController(e2e)-getRequestedAssessmentSummary",
       ]);
   });
 
+  it(
+    "Should not get the 'offering change' assessment requests details in the request summary " +
+      "for an eligible application when an eligible public institution user tries to access it and " +
+      "student offering change is pending with the ministry.",
+    async () => {
+      // Arrange
+
+      // Student has an application to the institution.
+      const student = await saveFakeStudent(db.dataSource);
+
+      const currentMSFAA = createFakeMSFAANumber(
+        { student },
+        {
+          state: MSFAAStates.Signed,
+        },
+      );
+      await db.msfaaNumber.save(currentMSFAA);
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          institution: collegeF,
+          institutionLocation: collegeFLocation,
+          student,
+          msfaaNumber: currentMSFAA,
+        },
+        { applicationStatus: ApplicationStatus.Completed },
+      );
+
+      // Create pending offering change.
+      const offeringRequestOfferings = createFakeOfferingRequestChange({
+        currentOffering: application.currentAssessment.offering,
+      });
+      await db.educationProgramOffering.save(offeringRequestOfferings);
+      const endpoint = `/institutions/assessment/student/${student.id}/application/${application.id}/requests`;
+      const institutionUserToken = await getInstitutionToken(
+        InstitutionTokenTypes.CollegeFUser,
+      );
+
+      // Act/Assert
+      await request(app.getHttpServer())
```

Comment:
My only concern about this type of test (where no result is expected) is that we need to be really sure that we have the proper data setup in place.
One idea would be calling the method `this.educationProgramOfferingService.getOfferingRequestsByApplicationId` to ensure that some data would be returned if the `showOfferingChange` was true, but I am not sure about the approach. Thoughts?

Reply:
One way would be setting up the test to get the details, assert that we get the details, change it in the DB and assert that we don't get the details. This way would be clear that it is because of the changes we did that we are getting a different response.

Reply:
The point about this endpoint is that the user does not have access to the "Offering Change Requests" at all.
It may work for other situations, but I am not sure if it would work for this specific one.

Reply:
IMO, this validation could be a part of test case at line 57 to return only valid assessment summary list by adding an offering change to the data and expect API response  not have requested offering change.

Reply:
Yes, i agree the part of this test case can return the list by adding the offerings in the data arrange.

Reply:
Yes, i agree the part of this test case can return the list by adding the offerings in the data arrange.

Reply:
Please keep in mind that the "Offering Change Requests" are executed by a different query on an isolated `if` condition. It is not about one query returning more or fewer records.

Reply:
as discussed adding ministry endpoint check, to validate the scenario

Reply:
@ann-aot I am ok with the approach, I am just leaving the comment open for the other DEVs have the opportunity to review also.

Reply:
> Offering Change Requests

@andrewsignori-aot not sure if this response was quoted for my comment. If so I was talking about expecting API result not having offering change in the array result, when there is an offering change for the application.

If not please ignore my comment :)